crates/swift-bridge-ir/src/bridged_type/bridgeable_custom_result.rs

```diff
@@ -0,0 +1,308 @@
+use crate::bridged_type::BridgedType;
+use crate::bridged_type::{BridgeableType, TypePosition, UnusedOptionNoneValue};
+use crate::{TypeDeclarations, SWIFT_BRIDGE_PREFIX};
+use proc_macro2::{Ident, Span, TokenStream};
+use quote::{format_ident, quote, ToTokens};
+use std::fmt::{Debug, Formatter};
+use syn::{Path, Type, TypeParam};
+
+#[derive(Clone)]
```

Comment:
I don't think that we should have a separate type for custom results.

Fundamentally, we're bridging a Result.

Sure, whether or not we need to emit a custom struct is dependent on the inner `ok_ty` and `err_ty` combination, but that can be handled inside of the `BuiltInResult`.

So, instead of having a `BuiltInResult` and a `CustomResult` we should just have a `BuiltInResult`, then when the `ok_ty` and `err_ty` combination requires a custom result FFI representation  we can emit a `#[repr(C)]` support type.

---

I know that in https://github.com/chinedufn/swift-bridge/issues/159#issuecomment-1438383178 I suggested a custom result. I was wrong.